examples/directml/stable_diffusion/stable_diffusion.py

```diff
@@ -257,15 +285,24 @@ def optimize(
     # Create a copy of the unoptimized model directory, then overwrite with optimized models from the olive cache.
     print("Copying optimized models...")
     shutil.copytree(unoptimized_model_dir, optimized_model_dir, ignore=shutil.ignore_patterns("weights.pb"))
-    for submodel_name in ("text_encoder", "vae_encoder", "vae_decoder", "safety_checker", "unet"):
+    for submodel_name in submodel_names:
         src_path = model_info[submodel_name]["optimized"]["path"]
         dst_path = optimized_model_dir / submodel_name / "model.onnx"
         shutil.copyfile(src_path, dst_path)
 
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser()
-    parser.add_argument("--model_id", default="runwayml/stable-diffusion-v1-5", type=str)
+    parser.add_argument(
+        "--model_id",
+        default="runwayml/stable-diffusion-v1-5",
+        choices=(
+            "CompVis/stable-diffusion-v1-4",
+            "runwayml/stable-diffusion-v1-5",
+            "sayakpaul/sd-model-finetuned-lora-t4",
+            "stabilityai/stable-diffusion-2",
+        ),
```

Comment:
This will prevent optimizing arbitrary LoRA models. It's probably not a big deal, but perhaps instead just warn the user if their choice isn't in this list of tested models.